v3/metadata/metadata.go

```diff
@@ -51,6 +59,34 @@ func Get(ctx context.Context, endpoint Endpoint) (string, error) {
 	return httpGet(ctx, url)
 }
 
+// FromCdRom retrieves metadata for Exoscale Private Instance,
+// from the attached CD-ROM(iso9660) device file system.
+// Important note: Run this code as privileged user.
+// Not Windows compatible.
+func FromCdRom(endpoint Endpoint) (string, error) {
+	disk, err := diskfs.Open(CdRomPath, diskfs.WithOpenMode(diskfs.ReadOnly))
+	if err != nil {
+		return "", fmt.Errorf("disk open: %w", err)
+	}
+
+	// TODO: Fix the block size in orchestrator from 512 to 2048
+	disk.DefaultBlocks = true
```

Comment:
Don't you need to close the disk at one point ?

Reply:
let me check with the diskfs lib thanks

Reply:
Tested successfully https://github.com/exoscale/egoscale/pull/634/commits/cddde218eb353ae2970f34d7ee98b583292c95f3 it the only way to close it, and it works